client/signup/navigation-link/index.jsx

```diff
@@ -68,7 +68,7 @@ const NavigationLink = React.createClass( {
 	},
 
 	render() {
-		if ( this.props.positionInFlow === 0 && this.props.direction === 'back' ) {
+		if ( this.props.positionInFlow === 0 && this.props.direction === 'back' && !this.props.stepSectionName ) {
```

Comment:
[Coding Guidelines](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md): `Any ! negation operator should have a following space`